apps/rpc/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import { env } from "@dotkomonline/env"
+import { type AppRouter, JwtService, appRouter, createContext } from "@dotkomonline/gateway-trpc"
+import fastifyCors from "@fastify/cors"
+import { type FastifyTRPCPluginOptions, fastifyTRPCPlugin } from "@trpc/server/adapters/fastify"
+import type { CreateFastifyContextOptions } from "@trpc/server/dist/adapters/fastify"
+import fastify from "fastify"
+
+const jwtService = new JwtService(env.WEB_AUTH0_ISSUER, [
+  env.WEB_AUTH0_CLIENT_ID,
+  env.DASHBOARD_AUTH0_CLIENT_ID,
+  env.GTX_AUTH0_CLIENT_ID,
+])
+
+const allowedOrigins = env.RPC_ALLOWED_ORIGINS.split(",")
+
+export async function createFastifyContext({ req }: CreateFastifyContextOptions) {
+  const bearer = req.headers.authorization
+  if (bearer !== undefined) {
+    const token = bearer.substring("Bearer ".length)
+    const principal = await jwtService.verify(token)
+    return createContext({ principal: principal.payload.sub ?? null })
+  }
+
+  return createContext({
+    principal: null,
+  })
+}
+
+const server = fastify({
+  maxParamLength: 5000,
+})
+server.register(fastifyCors, {
+  origin: allowedOrigins,
+  methods: ["GET", "POST", "PUT", "DELETE"],
+  allowedHeaders: ["Content-Type", "Authorization"],
+  credentials: true,
+})
+
+server.register(fastifyTRPCPlugin, {
+  prefix: "/api/trpc",
+  trpcOptions: {
+    router: appRouter,
+    createContext: createFastifyContext,
+    onError({ path, error }) {
+      // report to error monitoring
+      console.error(`Error in tRPC handler on path '${path}':`, error)
+    },
+  } satisfies FastifyTRPCPluginOptions<AppRouter>["trpcOptions"],
```

Comment:
Why is the `satisfies` needed? Isn't this already annotated from fastify's side?

Reply:
For some reason TS just leaves onError object as undefined. Alternative is doing `Parameters<OnErrorFunction<AppRouter, FastifyRequest>>[0]`

![image](https://github.com/user-attachments/assets/2078709c-626f-4ee1-ba05-44af1caf68f0)